konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/step/ml/classifier/ClassifierOutputStep.java

```diff
@@ -0,0 +1,100 @@
+
+
+@Builder
+@Data
+@Accessors(fluent = true)
+@AllArgsConstructor
+@JsonName("CLASSIFIER")
+@Schema(description = "ClassifierOutputStep takes as input a numerical 2d NDArray (i.e., float/double etc type) with shape [minibatch, numClasses] which represents the softmax predictions for a standard classifier and returns based on this array - The predicted class label - as a String\n" +
+        "- The predicted class index - as an integer (long)\n" +
+        "- The predicted class probability - as a Double")
+public class ClassifierOutputStep implements PipelineStep {
+    public static final String DEFAULT_OUTPUT_PROB_NAME = "prob";
+    public static final String DEFAULT_OUTPUT_INDEX_NAME = "index";
+    public static final String DEFAULT_OUTPUT_LABEL_NAME = "label";
+
+
+    @Schema(description = "inputName - optional. If set: this represents the NDArray. If not set: use DataUtils.inferField to find an NDArray field")
+    String inputName;
+
+
+    @Builder.Default
+    @Schema(description = "returnLabel, default is true; if false, don't return label")
+    boolean returnLabel = true;
+
+    @Builder.Default
+    @Schema(description = "returnIndex, default is true")
+    boolean returnIndex = true;
+
+    @Builder.Default
+    @Schema(description = " returnProb, default is true")
+    boolean returnProb = true;
+
+    @Builder.Default
+    @Schema(description = "output names for the labels")
+    String labelName = DEFAULT_OUTPUT_LABEL_NAME;
+
+    @Builder.Default
+    @Schema(description = "output names for the index")
+    String indexName = DEFAULT_OUTPUT_INDEX_NAME;
+
+    @Builder.Default
+    @Schema(description = "output names for the labels propabilities")
+    String probName = DEFAULT_OUTPUT_PROB_NAME;
+
+    @Schema(description = "as a List<String>. Optional. If not specified, the predicted class index as a string is used - i.e., \"0\", \"1\", etc")
+    List<String> Labels;
+
+    @Builder.Default
+    @Schema(description = "Integer, null by default. If non-null and > 1, we return List<String>, List<Long>, List<Double> for the predicted class/index/probability instead of String/Long/Double.")
+    Integer topN = null;
+
+    @Builder.Default
+    @Schema(description = "If true, also returns a List<List<Double>> of all probabilities (basically, converd NDArray to list. False by default.")
+    boolean allProbabilities = false;
+
+    public ClassifierOutputStep() {
+        //Normally this would be unnecessary to set default values here - but @Builder.Default values are NOT treated as normal default values.
+        //Without setting defaults here again like this, the fields would actually be null
+        this.returnLabel = true;
+        this.returnIndex = true;
+        this.returnProb = true;
+        this.indexName = DEFAULT_OUTPUT_INDEX_NAME;
+        this.labelName = DEFAULT_OUTPUT_LABEL_NAME;
+        this.probName = DEFAULT_OUTPUT_PROB_NAME;
+        this.topN = null;
+        this.allProbabilities = false;
```

Comment:
I've been a bit inconsistent on this (I need to do a pass to make this consistent throughout KS), but let's remove the `@Builder` and `@Builder.Default` (and the no-arg constructor) and rely on `@Accessors(fluent = true)` only.

In practice for users that means this:
```
ClassifierOutputStep.builder()
    .someMethod(...)
    .build()
```
becomes this:
```
new ClassifierOutputStep()
    .someMethod(...)
```

We avoid the lombok builder defaults issue with that too